### PR TITLE
Do not stop build process if some images are missing

### PR DESCRIPTION
**Pull request by llaville**
I've recently try to genereate [kanboard](https://github.com/kanboard/website) french documentation, and discover that some images are missing. 

Full generation was stop at first image missing. E.g: 

```
    Processing copy images for doc\fr_FR\html\index.html
    Processing copy images for doc\fr_FR\html\screenshots.html
file_get_contents(doc\fr_FR/screenshots/dropdown-screenshot.png): failed to open stream: No such file or directory
```

I'd like to have rather a warning message and continue to build all doc. Something like that 

```
  Applying Bookdown\Bookdown\Process\Resource\CopyImageProcess
    Processing copy images for doc\fr_FR\html\index.html
    Processing copy images for doc\fr_FR\html\screenshots.html
      Images doc\fr_FR/screenshots/dropdown-screenshot.png does not exists.
      Images doc\fr_FR/screenshots/task-screenshot.png does not exists.
```

### CODE REVIEW

File: src/Process/Resource/CopyImageProcess.php [hunk 1]
```diff
@@ -135,7 +135,12 @@ protected function downloadImage(DomNode $node)
         if (!$this->fsio->isDir($dir)) {
             $this->fsio->mkdir($dir);
         }
-        $this->fsio->put($file, $this->fsio->get($originFile));
+
+        if (file_exists($originFile)) {
+            $this->fsio->put($file, $this->fsio->get($originFile));
+        } else {
+            $this->logger->warning("      Images {$originFile} does not exists.");
```
Comment (sandrokeil) [inline src/Process/Resource/CopyImageProcess.php]: Please replace `Images` with `Image` and `exists` with `exist`. It's only one file.

**Comment by llaville** [comment]
Done. Thanks for code review !

**Comment by sandrokeil** [comment]
@llaville I will merge it soon. Thank you too. :+1:

**Comment by sandrokeil** [comment]
@llaville I think it's better here to use a try/catch instead of `file_exists`, because the download of the file can fail. What do you think?

**Comment by llaville** [comment]
@sandrokeil  Sorry to reply so lately - was very busy on office with different projects. 

I don't see in what case we should need a try/catch - could you explain a real use case please ?

**Comment by sandrokeil** [comment]
@llaville We throw an exception [here](https://github.com/bookdown/Bookdown.Bookdown/blob/1.x/src/Fsio.php#L17) if `file_get_contents` failed. It's an edge case that `file_exists` returns true and `file_get_contents` false, but it can happen. If we use a try/catch, we can fix/avoid this.

**Comment by llaville** [comment]
@sandrokeil Forgotten your comment; Thanks for it. I've just applied the patch following your comment !

**Pull request merged**